checkpoint.go

```diff
@@ -78,12 +79,30 @@ checkpointed.`,
 	},
 }
 
-func getCheckpointImagePath(context *cli.Context) string {
+func prepareImagePaths(context *cli.Context) (string, string, error) {
 	imagePath := context.String("image-path")
+
 	if imagePath == "" {
 		imagePath = getDefaultImagePath(context)
 	}
-	return imagePath
+
+	if err := os.MkdirAll(imagePath, 0755); err != nil {
+		return "", "", err
+	}
+
+	parentPath := context.String("parent-path")
+
+	if parentPath == "" {
+		return imagePath, parentPath, nil
+	}
+
+	realParent := filepath.Join(imagePath, parentPath)
+	if fi, err := os.Stat(realParent); err != nil || !fi.IsDir() {
+		return "", "", fmt.Errorf("--parent-path needs to be relative path of --image-path, and points to previous checkpoint directory")
```

Comment:
I think this belongs to option description, in here you can just return `fmt.Errorf("invalid --parent-path %s: %w", parentPath, err)` or something like that.

Reply:
...with
```go
if err == nil && !fi.IsDir {
   err := &os.PathError(Path: realParent, Err: unix.ENOTDIR)
}
```